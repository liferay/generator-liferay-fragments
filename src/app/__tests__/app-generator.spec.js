const fs = require('fs');
const glob = require('glob');
const path = require('path');
const YeomanTest = require('yeoman-test');

/**
 * @param {string} base
 * @param {string} _path
 */
function expectFile(base, _path) {
  const filePath = path.join(base, _path);
  const stat = fs.lstatSync(filePath);
  let content = '';

  if (stat.isFile()) {
    content = fs.readFileSync(filePath, 'utf-8');
  }

  return expect({
    filePath: _path.split(path.sep).join('/'),
    content,
  }).toMatchSnapshot();
}

/**
 * @param {string} base
 * @param {string[]} paths
 */
function expectFiles(base, paths) {
  return paths.map((_path) => expectFile(base, _path));
}

/**
 * @param {string} base
 */
function expectProjectFiles(base) {
  const templatesPath = path.resolve(__dirname, '..', 'templates');

  return expectFiles(
    base,
    glob.sync(`${templatesPath}/**/*`, { dot: true }).map((templatePath) =>
      path
        .resolve(templatePath)
        .replace(templatesPath, '')
        .replace(/\.ejs$/i, '')
    )
  );
}

describe('app-generator', () => {
  it('generates a new project', () =>
    YeomanTest.run(path.join(__dirname, '..')).then((projectPath) =>
      expectProjectFiles(path.join(projectPath, 'sample-liferay-fragments'))
    ));

  it('allows a custom repository name', () =>
    YeomanTest.run(path.join(__dirname, '..'))
      .withPrompts({ projectName: 'My Nice Custom Project' })
      .then((projectPath) =>
        expectProjectFiles(path.join(projectPath, 'my-nice-custom-project'))
      ));

  it('allows adding sample content', () =>
    YeomanTest.run(path.join(__dirname, '..'))
      .withPrompts({ addSampleContent: true })
      .then((projectPath) => {
        const contentPath = path.join(projectPath, 'sample-liferay-fragments');

        expectProjectFiles(contentPath);

        expectFiles(
          path.join(contentPath, 'src', 'sample-collection', 'sample-fragment'),
          ['fragment.json', 'index.html', 'styles.css', 'main.js']
        );
      }));
});
