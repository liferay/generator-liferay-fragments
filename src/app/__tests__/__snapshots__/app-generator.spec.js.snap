// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-generator allows a custom repository name 1`] = `
Object {
  "content": "root = true

[*]
indent_style = space
indent_size = 2
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true

[*.md]
trim_trailing_whitespace = false
",
  "filePath": "/.editorconfig",
}
`;

exports[`app-generator allows a custom repository name 2`] = `
Object {
  "content": "build/
node_modules/
.yo-rc.json
",
  "filePath": "/.gitignore",
}
`;

exports[`app-generator allows a custom repository name 3`] = `
Object {
  "content": "{}
",
  "filePath": "/.yo-rc.json",
}
`;

exports[`app-generator allows a custom repository name 4`] = `
Object {
  "content": "{
  \\"name\\": \\"my-nice-custom-project\\",
  \\"description\\": \\"Liferay Fragments project\\",
  \\"version\\": \\"1.0.0\\",

  \\"engines\\": {
    \\"node\\": \\">= 8.0.0\\",
    \\"npm\\": \\">= 6.0.0\\"
  },

  \\"keywords\\": [
    \\"liferay\\",
    \\"liferay-fragments\\"
  ],

  \\"scripts\\": {
    \\"add-collection\\": \\"yo liferay-fragments:collection\\",
    \\"add-fragment\\": \\"yo liferay-fragments:fragment\\",
    \\"add-fragment-composition\\": \\"yo liferay-fragments:fragment-composition\\",
    \\"add-page-template\\": \\"yo liferay-fragments:page-template\\",
    \\"compress\\": \\"yo liferay-fragments:compress\\",
    \\"export\\": \\"yo liferay-fragments:export\\",
    \\"import\\": \\"yo liferay-fragments:import\\",
    \\"import:watch\\": \\"yo liferay-fragments:import --watch\\",
    \\"preview\\": \\"yo liferay-fragments:preview\\"
  }
}
",
  "filePath": "/package.json",
}
`;

exports[`app-generator allows a custom repository name 5`] = `
Object {
  "content": "# My Nice Custom Project

This project has been created with [generator-liferay-fragments][1]. You can read
how to manage this project in the [generator documentation][2].

[1]: https://www.npmjs.com/package/generator-liferay-fragments
[2]: https://www.npmjs.com/package/generator-liferay-fragments#usage
",
  "filePath": "/README.md",
}
`;

exports[`app-generator allows a custom repository name 6`] = `
Object {
  "content": "",
  "filePath": "/src",
}
`;

exports[`app-generator allows a custom repository name 7`] = `
Object {
  "content": "",
  "filePath": "/src/.gitkeep",
}
`;

exports[`app-generator allows adding sample content 1`] = `
Object {
  "content": "root = true

[*]
indent_style = space
indent_size = 2
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true

[*.md]
trim_trailing_whitespace = false
",
  "filePath": "/.editorconfig",
}
`;

exports[`app-generator allows adding sample content 2`] = `
Object {
  "content": "build/
node_modules/
.yo-rc.json
",
  "filePath": "/.gitignore",
}
`;

exports[`app-generator allows adding sample content 3`] = `
Object {
  "content": "{}
",
  "filePath": "/.yo-rc.json",
}
`;

exports[`app-generator allows adding sample content 4`] = `
Object {
  "content": "{
  \\"name\\": \\"sample-liferay-fragments\\",
  \\"description\\": \\"Liferay Fragments project\\",
  \\"version\\": \\"1.0.0\\",

  \\"engines\\": {
    \\"node\\": \\">= 8.0.0\\",
    \\"npm\\": \\">= 6.0.0\\"
  },

  \\"keywords\\": [
    \\"liferay\\",
    \\"liferay-fragments\\"
  ],

  \\"scripts\\": {
    \\"add-collection\\": \\"yo liferay-fragments:collection\\",
    \\"add-fragment\\": \\"yo liferay-fragments:fragment\\",
    \\"add-fragment-composition\\": \\"yo liferay-fragments:fragment-composition\\",
    \\"add-page-template\\": \\"yo liferay-fragments:page-template\\",
    \\"compress\\": \\"yo liferay-fragments:compress\\",
    \\"export\\": \\"yo liferay-fragments:export\\",
    \\"import\\": \\"yo liferay-fragments:import\\",
    \\"import:watch\\": \\"yo liferay-fragments:import --watch\\",
    \\"preview\\": \\"yo liferay-fragments:preview\\"
  }
}
",
  "filePath": "/package.json",
}
`;

exports[`app-generator allows adding sample content 5`] = `
Object {
  "content": "# Sample Liferay Fragments

This project has been created with [generator-liferay-fragments][1]. You can read
how to manage this project in the [generator documentation][2].

[1]: https://www.npmjs.com/package/generator-liferay-fragments
[2]: https://www.npmjs.com/package/generator-liferay-fragments#usage
",
  "filePath": "/README.md",
}
`;

exports[`app-generator allows adding sample content 6`] = `
Object {
  "content": "",
  "filePath": "/src",
}
`;

exports[`app-generator allows adding sample content 7`] = `
Object {
  "content": "",
  "filePath": "/src/.gitkeep",
}
`;

exports[`app-generator allows adding sample content 8`] = `
Object {
  "content": "{
  \\"cssPath\\": \\"styles.css\\",
  \\"configurationPath\\": \\"configuration.json\\",
  \\"htmlPath\\": \\"index.html\\",
  \\"jsPath\\": \\"main.js\\",
  \\"name\\": \\"Sample fragment\\",
  \\"type\\": \\"section\\"
}
",
  "filePath": "fragment.json",
}
`;

exports[`app-generator allows adding sample content 9`] = `
Object {
  "content": "<div class=\\"sample-fragment\\">
  Sample fragment
</div>
",
  "filePath": "index.html",
}
`;

exports[`app-generator allows adding sample content 10`] = `
Object {
  "content": ".sample-fragment {
  color: inherit;
  font-size: 18px;
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI',
    'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans',
    'Helvetica Neue', sans-serif;
}
",
  "filePath": "styles.css",
}
`;

exports[`app-generator allows adding sample content 11`] = `
Object {
  "content": "console.group('Sample fragment');
console.log('fragmentElement', fragmentElement);
console.groupEnd();
",
  "filePath": "main.js",
}
`;

exports[`app-generator generates a new project 1`] = `
Object {
  "content": "root = true

[*]
indent_style = space
indent_size = 2
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true

[*.md]
trim_trailing_whitespace = false
",
  "filePath": "/.editorconfig",
}
`;

exports[`app-generator generates a new project 2`] = `
Object {
  "content": "build/
node_modules/
.yo-rc.json
",
  "filePath": "/.gitignore",
}
`;

exports[`app-generator generates a new project 3`] = `
Object {
  "content": "{}
",
  "filePath": "/.yo-rc.json",
}
`;

exports[`app-generator generates a new project 4`] = `
Object {
  "content": "{
  \\"name\\": \\"sample-liferay-fragments\\",
  \\"description\\": \\"Liferay Fragments project\\",
  \\"version\\": \\"1.0.0\\",

  \\"engines\\": {
    \\"node\\": \\">= 8.0.0\\",
    \\"npm\\": \\">= 6.0.0\\"
  },

  \\"keywords\\": [
    \\"liferay\\",
    \\"liferay-fragments\\"
  ],

  \\"scripts\\": {
    \\"add-collection\\": \\"yo liferay-fragments:collection\\",
    \\"add-fragment\\": \\"yo liferay-fragments:fragment\\",
    \\"add-fragment-composition\\": \\"yo liferay-fragments:fragment-composition\\",
    \\"add-page-template\\": \\"yo liferay-fragments:page-template\\",
    \\"compress\\": \\"yo liferay-fragments:compress\\",
    \\"export\\": \\"yo liferay-fragments:export\\",
    \\"import\\": \\"yo liferay-fragments:import\\",
    \\"import:watch\\": \\"yo liferay-fragments:import --watch\\",
    \\"preview\\": \\"yo liferay-fragments:preview\\"
  }
}
",
  "filePath": "/package.json",
}
`;

exports[`app-generator generates a new project 5`] = `
Object {
  "content": "# Sample Liferay Fragments

This project has been created with [generator-liferay-fragments][1]. You can read
how to manage this project in the [generator documentation][2].

[1]: https://www.npmjs.com/package/generator-liferay-fragments
[2]: https://www.npmjs.com/package/generator-liferay-fragments#usage
",
  "filePath": "/README.md",
}
`;

exports[`app-generator generates a new project 6`] = `
Object {
  "content": "",
  "filePath": "/src",
}
`;

exports[`app-generator generates a new project 7`] = `
Object {
  "content": "",
  "filePath": "/src/.gitkeep",
}
`;
